src/snitch_string_utility.cpp

```diff
@@ -4,6 +4,55 @@
 #include <cstring> // for std::memcpy
```

Comment:
nit: I'd add the used headers here (`<string_view>`, `snitch_error_handling.hpp`)

Reply:
I added the include to `snitch_error_handling.hpp`, thanks. However I left out `<string_view>` since it is already included in the header.

I also noticed I was incorrectly using `terminate_with()` instead of `assertion_failed()`, and fixed that in the process.